driver-sync/src/test/functional/com/mongodb/client/ClientEncryptionCustomEndpointTest.java

```diff
@@ -237,6 +240,19 @@ public void testCustomEndpoint() {
                         + "  \"endpoint\": \"example.com:443\"\n"
                         + "}"),
                 false, MongoClientException.class, MongoCryptException.class, "Invalid KMS response"});
+        data.add(new Object[]{"10. [kmip] default endpoint",
+                "kmip",
+                BsonDocument.parse("{\n"
+                        + "  \"keyId\": \"1\"\n"
+                        + "}"),
+                false, null, null, null});
+        data.add(new Object[]{"10. [kmip] default endpoint",
+                "kmip",
+                BsonDocument.parse("{\n"
+                        + "  \"keyId\": \"1\",\n"
+                        + "  \"endpoint\": \"localhost:5698\"\n"
+                        + "}"),
+                false, null, null, null});
```

Comment:
I have added another test case 12. I think it will resemble:

```
data.add(new Object[]{"12. [kmip] invalid endpoint from DataKeyOptions",
                "kmip",
                BsonDocument.parse("{\n"
                        + "  \"keyId\": \"1\",\n"
                        + "  \"endpoint\": \"doesnotexist.local:5698\"\n"
                        + "}"),
                false, /* TODO */, /* TODO */, "TODO"});
```

Reply:
Done